src/Raven.Client/Documents/Session/Operations/BatchOperation.cs

```diff
@@ -64,48 +70,192 @@ CommandType GetCommandType(BlittableJsonReaderObject batchResult)
                         $"Cluster transaction was send to a node that is not supporting it. So it was executed ONLY on the requested node on {_session.RequestExecutor.Url}.");
             }
 
-            for (var i = 0; i < _sessionCommandsCount; i++)
+            for (var i = 0; i < _allCommandsCount; i++)
```

Comment:
Better to have two `for` statements, one from `0 .. _sessionCommandsCount` and one from `_sessionCommandsCount` to `_allCommandsCount`

Reply:
Done